client/my-sites/activity/filterbar/index.jsx

```diff
@@ -74,8 +74,30 @@ export class Filterbar extends Component {
 		}
 	};
 
+	isEmptyFilter = filter => {
+		if ( ! filter ) {
+			return true;
+		}
+		if ( filter.group || filter.on || filter.before || filter.after ) {
+			return false;
+		}
+		if ( filter.page !== 1 ) {
+			return false;
+		}
+		return true;
+	};
+
 	render() {
-		const { translate, siteId, filter } = this.props;
+		const { translate, siteId, filter, isLoading, isVisible } = this.props;
+
+		if ( siteId && isLoading && this.isEmptyFilter( filter ) ) {
+			return <div className="filterbar is-loading" />;
+		}
+
+		if ( ! isVisible ) {
```

Comment:
shouldn't this check go first?

Reply:
nope since we don't know about this part yet. Since the data is still loading. 
We don't show the Filterbar if there is nothing to filter.

Reply:
Will add a comment.